rules/Transform/Rector/Class_/AddAllowDynamicPropertiesAttributeRector.php

```diff
@@ -139,16 +139,18 @@ private function addAllowDynamicPropertiesAttribute(Class_ $class): Class_
 
     private function shouldSkip(Class_ $class): bool
     {
-        if ($this->transformOnNamespaces !== []) {
-            $className = (string) $this->nodeNameResolver->getName($class);
-            foreach ($this->transformOnNamespaces as $transformOnNamespace) {
-                if ($this->nodeNameResolver->isStringName($className, $transformOnNamespace)) {
-                    continue;
-                }
-
+        $className = (string) $this->nodeNameResolver->getName($class);
+        foreach ($this->transformOnNamespaces as $transformOnNamespace) {
+            if (str_contains($transformOnNamespace, '*')) {
                 if (! fnmatch($transformOnNamespace, $className, FNM_NOESCAPE)) {
                     return true;
```

Comment:
@TomasVotruba @staabm this conditions seems need to be on last resort to check because it returns true on negation early, I will create separate PR to cover too late check.